### PR TITLE
Add dockerfile and script to build cuDF jar

### PR DESCRIPTION
**Pull request by GaryShen2008**
This commit is to build cuDF jar file in a Centos7 docker container.
Provide a dockerfile, a build script and a README to explain the steps.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: java/ci/Dockerfile.centos7
Comment (jlowe) [inline java/ci/Dockerfile.centos7:1]: Missing the copyright and license comment block

File: java/ci/build-in-docker.sh
Comment (jlowe) [inline java/ci/build-in-docker.sh:2]: Missing the copyright and license comment block

File: java/ci/build-in-docker.sh
Comment (jlowe) [inline java/ci/build-in-docker.sh:34]: Per an offline discussion with @kkraus14 we shouldn't need to explicitly off the CXX11 ABI, as devtoolset is already building binaries for an OS that's older than that ABI change.

This cmake option is being actively removed from the projects, so it should be removed from this cmake command along with the ones down below.

Comment (kkraus14) [reply java/ci/build-in-docker.sh:34]: Working on removing it in #6209

**Review by dillon-cullinan — COMMENTED** [review]
ops-codeowners should not be getting picked up here. Can you edit the `CODEOWNERS` file to address this? Thank you!

This should probably be removed: https://github.com/rapidsai/cudf/blob/branch-0.16/.github/CODEOWNERS#L20

**Comment by ajschmidt8** [comment]
> ops-codeowners should not be getting picked up here. Can you edit the `CODEOWNERS` file to address this? Thank you!
> 
> This line should probably be removed: https://github.com/rapidsai/cudf/blob/branch-0.16/.github/CODEOWNERS#L20

This was addressed in #6212. No need for any further action here 👍

File: java/ci/Dockerfile.centos7
Comment (GaryShen2008) [reply java/ci/Dockerfile.centos7:17]: Added the copyright.

File: java/ci/build-in-docker.sh
Comment (GaryShen2008) [reply java/ci/build-in-docker.sh:2]: Added the copyright.

File: java/ci/build-in-docker.sh
Comment (GaryShen2008) [reply java/ci/build-in-docker.sh]: Wait for #6209. And need to remove abiOff profile in pom.

**Review by jlowe — APPROVED** [review]
Looks good to me.  We can wait until #6209 is merged.  I'll put up a PR later today to remove the cxx11abi setting from the pom and Java native cmake files.

File: java/ci/build-in-docker.sh
Comment (jlowe) [reply java/ci/build-in-docker.sh]: I filed #6223 to remove the CXX11 ABI flag from the pom and JNI cmake.

File: java/ci/build-in-docker.sh
Comment (jlowe) [inline java/ci/build-in-docker.sh:71]: The `abiOff` profile was removed in #6223 so we should remove it here as well.  Otherwise the Maven build will complain about an unrecognized profile.

**Review by jlowe — CHANGES_REQUESTED** [review]
@GaryShen2008 After removing the reference to the removed `abiOff` profile I think this will be ready to go.

**Review by jlowe — APPROVED** [review]

**Pull request merged**
